plugins/git2/git2.plugin.zsh

```diff
@@ -0,0 +1,332 @@
+compdef _git gPd=git-push
+# git push force
+alias gPf='git push --force'
+compdef _git gPf=git-push
+
+################
+# Git Fetch (gf)
+################
+# Git fetch current branch
+alias gf='git fetch'
+compdef _git gf='git-fetch'
+# Git fetch all branches
+alias gfa='git fetch --all'
+compdef _git gfa='git-fetch'
+# reset your work to the latest status of the code base (and fetch all other branches).
+# (git fetch all branch and rebase)
+alias gfagpr='git fetch --all && git pull --rebase'
+compdef _git gfagpr='git-pull'
+
+#################
+# Git Commit (gc)
+#################
+# Git commit
+alias gc='git commit -v'
+compdef _git gc=git-commit
+# Git commit amend
+alias gc!='git commit -v --amend'
+compdef _git gc!=git-commit
+# Git commit all tracked files
+alias gca='git commit -v -a'
+compdef _git gca=git-commit
+# Git commit amend all tracked files
+alias gca!='git commit -v -a --amend'
+compdef _git gca!=git-commit
+# Git commit with message
+alias gcmsg='git commit -m'
+compdef _git gcmsg=git-commit
+
+####################
+# Git Checkout (gco)
+####################
+# git checkout
+alias gco='git checkout'
+compdef _git gco=git-checkout
+# Git checkout master
+alias gcom='git checkout master'
+compdef _git gcom=git-checkout
+
+# Git Remote (gre)
+alias gre='git remote'
+compdef _git gre=git-remote
+alias grev='git remote -v'
+compdef _git grv=git-remote
+alias gremv='git remote rename'
+compdef _git gremv=git-remote
+alias grerm='git remote remove'
+compdef _git grrm=git-remote
+alias greset='git remote set-url'
+compdef _git greset=git-remote
+alias greup='git remote update'
+compdef _git greset=git-remote
+
+# Git Rebase (gr)
+alias gr='git rebase -i'
+compdef _git gr=git-rebase
+alias grc='git rebase --continue'
+compdef _git grc=git-rebase
+alias gra='git rebase --abort'
+compdef _git gra=git-rebase
+alias grs='git rebase --skip'
+compdef _git grs=git-rebase
+alias grm='git rebase master'
+compdef _git grm=git-rebase
+
+#####################################################
+# Git rebase Interactively N commit (rewrite history)
+#####################################################
+alias gr2='git rebase -i HEAD~2'
+compdef _git gr2=git-rebase
+alias gr3='git rebase -i HEAD~3'
+compdef _git gr3=git-rebase
+alias gr4='git rebase -i HEAD~4'
+compdef _git gr4=git-rebase
+alias gr5='git rebase -i HEAD~5'
+compdef _git gr5=git-rebase
+alias gr6='git rebase -i HEAD~6'
+compdef _git gr6=git-rebase
+alias gr7='git rebase -i HEAD~7'
+compdef _git gr7=git-rebase
+alias gr8='git rebase -i HEAD~8'
+compdef _git gr8=git-rebase
+alias gr9='git rebase -i HEAD~9'
+compdef _git gr9=git-rebase
+alias gr10='git rebase -i HEAD~10'
+compdef _git gr10=git-rebase
+
+# Git Branch (gb)
+alias gb='git branch'
+compdef _git gb=git-branch
+alias gba='git branch -a'
+compdef _git gba=git-branch
+
+##############
+# Git Log (gl)
+##############
+alias gl='git log'
+compdef _git gl=git-log
+# Git log graph
+alias glg='git log --stat --max-count=10'
```

Comment:
Do we really need `--max-count=10`? I want to get rid of this from all `git log` aliases.